### PR TITLE
Second milestone User interface completed

### PR DESCRIPTION
**Pull request by xtrmdarc**
We kindly request a code review for our second milestone project.

- Added methods for user communication.
- Mock workflow for the tic tac toe game.

Thank you.

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -1,3 +1,69 @@
...
+}
+
+board = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
+
+current_player = player1
+puts UserInterface.display_board(board)
+
+loop do
+  # We are only doing 5 moves for demo purposes.
+  5.times do
```
Comment (Forison) [inline bin/main.rb]: [ ]  it better to use a terminating variable as below‎👇. instead of ```5.times do ... end```
```
while game_on
 end
```

**Review by Forison — CHANGES_REQUESTED** [review]
## Status: Require changes.🛑
# 🙋‍♂️ Team.
## Here are a few issues we have to address to get this project approved🙂.
* [x] The game should ask players for their names ([screenshot hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/logic.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [x] The game should repeat all actions for the next player's move.([screenshot hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/logic.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
 Currently, the game ends after six turns total, 3 turns each from each player. In other words, a winner is declared after the sixth move and then the game ends even when the sixth move is not a winning move.
## That is it, Kindly make these changes and submit the project for review.
#### Happy coding 💻
#### Addo Forison

**Comment by xtrmdarc** [comment]
Thank you, Addo for your feedback.
We are asking for player names now as you requested and we clarified the application workflow. We are not running the game loop only 5 times now, we are running it indefinitely. Of course, the game is still not functional so the game state will never be validated and the loop will continue indefinitely.

**Review by fatymahmed — APPROVED** [review]
### Hi @xtrmdarc @ivanid22,
### Great job making the suggested changes🎉🎉
[OPTIONAL]
- Your game should inform players if their move is invalid. Please add this to your game flow before merging, I marked it as optional since it was missed in the first review.

### It's a 👍🏾 from me!
## ✔️ Approved!

Good luck with your next projects.
 _Don't forget to submit your project completion form_

**Pull request merged**